tornado/autoreload.py

```diff
@@ -209,7 +209,10 @@ def _reload():
     global _reload_attempted
     _reload_attempted = True
     for fn in _reload_hooks:
-        fn()
+        try:
+            fn()
+        except Exception as e:
+            gen_log.warning("Error occured on reload: %s" % e)
```

Comment:
This will allow the autoreload to work but is not ideal - it will print a warning on every reload for apps that use asyncio, but this is perfectly expected and should be silent. The reload hook should use a wrapper function so that we silence all exceptions from IOLoop.close. 

When we have that, I'm not sure that it makes sense to keep this part of the change, but if we do it should use `app_log.error` instead of `gen_log.warning` because it's potentially an error in application code.